### PR TITLE
Panel: Add design documentation to readme

### PR DESCRIPTION
**Pull request by davewhitley**
Initial commit to update Panel readme. These changes add design documentation.

## Description

This PR adds design guideline documentation in addition to the development documentation that existed previously. These guidelines are 'best practices' for the usage of the component, as well as describing the component in more detail.

## Screenshots

A preview can be seen here, although it doesn't represent what it will look like in the handbook:

[preview](https://github.com/WordPress/gutenberg/blob/96948ffccc863d1d51b7ed0753864e3262b47e94/packages/components/src/panel/README.md)

## Feedback

I'd appreciate feedback on:

- Whether my markdown syntax is correct
- Design guidelines
- Double-checking a11y of the document
- Seeing if the design guidelines align with the development documentation

Thank you!

### CODE REVIEW

**Comment by drw158** [comment]
For the dev documentation I moved the code block up, and I had to update the heading levels.

File: packages/components/src/panel/README.md [hunk 1]
```diff
@@ -1,71 +1,144 @@
...
+
+1. [Design guidelines](#design-guidelines)
+2. [Development guidelines](#development-guidelines)
+3. [Related components](#related-components)
+
+## Design guidelines
+
+### Anatomy
+
+A Panel consists of a single continuous column of subdivisions that contain items of content.
```
Comment (kjellr) [inline packages/components/src/panel/README.md]: Reading this leads me to believe the panel is the entire column, but it's actually just the individual expandable/collapsible section, right? I wonder if we could rephrase this to make that clear. Maybe something like:

> A Panel is a single section of content that can be expanded or collapsed as needed.

**Review by kjellr — APPROVED** [review]
Looks great, thanks @drw158! 🚢
